rust/arrow/src/array.rs

```diff
@@ -627,6 +627,26 @@ impl<'a> From<Vec<&'a str>> for BinaryArray {
     }
 }
 
+impl<'a> From<Vec<&[u8]>> for BinaryArray {
+    fn from(v: Vec<&[u8]>) -> Self {
+        let mut offsets = vec![];
```

Comment:
It would be more efficient to initialize these vectors with `Vec::with_capacity(v.len())`